frontend/src/queries/nodes/DataTable/dataTableLogic.ts

```diff
@@ -1,32 +1,93 @@
+        }
+        return props.key
+    }),
     path(['queries', 'nodes', 'DataTable', 'dataTableLogic']),
     actions({ setColumns: (columns: HogQLExpression[]) => ({ columns }) }),
     reducers(({ props }) => ({
         columns: [getColumnsForQuery(props.query), { setColumns: (_, { columns }) => columns }],
     })),
-    connect({
-        values: [featureFlagLogic, ['featureFlags']],
-    }),
+    connect((props: DataTableLogicProps) => ({
+        values: [
+            featureFlagLogic,
+            ['featureFlags'],
+            dataNodeLogic({ key: props.key, query: props.query.source }),
+            ['response', 'responseLoading'],
+        ],
+    })),
     selectors({
+        sourceKind: [(_, p) => [p.query], (query): NodeKind | null => query.source?.kind],
+        orderBy: [
+            (_, p) => [p.query],
+            (query): string[] | null => (isEventsQuery(query.source) ? query.source.orderBy || ['-timestamp'] : null),
+            { resultEqualityCheck: objectsEqual },
+        ],
+        resultsWithCategoryRows: [
+            (s) => [s.sourceKind, s.orderBy, s.response, s.columns],
+            (sourceKind, orderBy, response, columns): any[] | null => {
+                if (sourceKind === NodeKind.EventsQuery) {
+                    const results = (response as EventsQuery['response'] | null)?.results
+                    if (results) {
+                        const orderKey =
+                            (orderBy?.[0]?.startsWith('-') ? orderBy[0].slice(1) : orderBy?.[0]) || 'timestamp'
+                        if (orderKey === 'timestamp') {
+                            const orderKeyIndex = columns.findIndex(
+                                (column) =>
+                                    removeExpressionComment(column) === orderKey ||
+                                    removeExpressionComment(column) === `-${orderKey}`
+                            )
+                            if (orderKeyIndex !== -1) {
+                                let lastResult: any | null = null
+                                const newResults: any[] = []
+                                for (const result of results) {
+                                    if (
+                                        result &&
+                                        lastResult &&
+                                        !dayjs(result[orderKeyIndex]).isSame(lastResult[orderKeyIndex], 'day')
+                                    ) {
+                                        newResults.push({
+                                            [categoryRowKey]: dayjs(result[orderKeyIndex]).format('LL'),
+                                        })
+                                    }
+                                    newResults.push(result)
+                                    lastResult = result
+                                }
+                                return newResults
```

Comment:
I may be being dense, but I don't quite see why we're doing this at a glance. Can you add a comment for the future traveler?

Reply:
To get those date rows that you blocked me on before :D... or what do you mean?

Reply:
I renamed it to `resultsWithLabelRows` and added a comment.